src/util/mod.rs

```diff
@@ -523,6 +595,57 @@ mod tests {
     }
 
     #[test]
+    fn test_ring_queue_to_vec() {
+        let cap = 10;
+        for pos in 0..cap {
+            let mut queue = RingQueue::with_capacity(cap);
+            for num in 0..cap + pos {
+                queue.push(num);
+            }
+            let exp: Vec<_> = (0..cap + pos).collect();
+            // now queue's start index is at pos.
+            let v = queue.to_vec(0..queue.len());
+            assert_eq!(*v, exp[pos..]);
+            for i in 0..cap {
+                queue.drain(cap - i - 1..cap - i);
+                let res = queue.to_vec(0..queue.len());
+                assert_eq!(*res, exp[pos..pos + cap - i - 1]);
```

Comment:
I thought it would be easier to understand to initialize `exp` to be `(pos.. pos +cap)` and to use it with start index set to 0.